#! /bin/sh -e
## 28_cobalt_ide.dpatch by Peter Horton

## DP: Fix long IDE detection delay by not scanning non-existent channels.
## DP: Not quite the right fix according to Alan Cox but it works(tm).


if [ $# -lt 1 ]; then
    echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch} ${2:+-d $2}"

case "$1" in
    -patch) patch -p1 ${patch_opts} < $0;;
    -unpatch) patch -R -p1 ${patch_opts} < $0;;
    *)
        echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
        exit 1;;
esac

exit 0


Index: linux.git/include/asm-mips/mach-cobalt/ide.h
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ linux.git/include/asm-mips/cobalt/ide.h	2006-01-22 23:17:03.000000000 +0000
@@ -0,0 +1,8 @@
+#ifndef __ASM_COBALT_IDE_H
+#define __ASM_COBALT_IDE_H
+
+#define MAX_HWIFS		2
+
+#include <asm/mach-generic/ide.h>
+
+#endif

